lib/views/chatroom/chatroom_components/chat_bar.dart

```diff
@@ -0,0 +1,262 @@
+import 'package:custom_pop_up_menu/custom_pop_up_menu.dart';
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'package:google_fonts/google_fonts.dart';
+import 'package:group_chat_example/constants.dart';
+import 'package:group_chat_example/utils/ui_utils.dart';
+import 'package:group_chat_example/views/chatroom/views/chatroom_polls_page.dart';
+
+class ChatBar extends StatefulWidget {
+  const ChatBar({super.key});
+
+  @override
+  State<ChatBar> createState() => _ChatBarState();
+}
+
+class _ChatBarState extends State<ChatBar> {
+  late CustomPopupMenuController _popupMenuController;
+  late TextEditingController _textEditingController;
```

Comment:
Remember to dispose of the TextEditingController when it is no longer needed. This will ensure we discard any resources used by the object.

Reply:
Oh yes, true